akd_local_auditor/src/storage/s3/test.rs

```diff
@@ -0,0 +1,368 @@
+        endpoint: Some(TEST_ENDPOINT.to_string()),
+        access_key: Some(ACCESS_KEY.to_string()),
+        secret_key: Some(SECRET_KEY.to_string()),
+    };
+    let storage: S3AuditStorage = (&clap_args).into();
+    storage
+}
+
+// =========================== Test Cases =========================== //
+
+#[tokio::test]
+#[ignore]
+async fn integration_test_bucket_listing() -> Result<()> {
+    // make sure we have a valid bucket name, that's "somewhat" unique
+    let bucket = bucket_name!();
+    log::debug!("Test bucket name is {}", bucket);
+    assert!(matches!(is_bucket_name_valid(&bucket), Ok(_)));
+
+    // Get the storage reader
+    let storage = get_s3_storage(&bucket);
+    let shared_config = storage.get_shared_config().await;
+
+    // Populate the test storage
+    populate_test_storage(&shared_config, &bucket, 10).await?;
+
+    // List the epochs found in the storage layer
+    let mut epoch_summaries: Vec<EpochSummary> = storage.list_proofs().await?;
+    epoch_summaries.sort_by(|a, b| a.name.epoch.cmp(&b.name.epoch));
+
+    // There should be 10 proofs in the storage layer
+    log::info!(
+        "There are {} epochs in the storage layer",
+        epoch_summaries.len()
+    );
+    assert_eq!(10, epoch_summaries.len());
+
+    // check the linear history of the proofs
+    log::info!("Checking linear history of audit proofs");
+    let mut i = 0u64;
+    for summary in epoch_summaries {
+        assert_eq!(i, summary.name.epoch);
+        i += 1;
+    }
+
+    // if the test is successful, try a cleanup of the storage now
+    maybe_flush_storage(&shared_config, &bucket).await?;
+
+    Ok(())
+}
+
+#[tokio::test]
+#[ignore]
+async fn integration_test_audit_verification() -> Result<()> {
+    // make sure we have a valid bucket name, that's "somewhat" unique
+    let bucket = bucket_name!();
+    log::debug!("Test bucket name is {}", bucket);
+    assert!(matches!(is_bucket_name_valid(&bucket), Ok(_)));
+
+    // Get the storage reader
+    let storage = get_s3_storage(&bucket);
+    let shared_config = storage.get_shared_config().await;
+
+    // Populate the test storage
+    populate_test_storage(&shared_config, &bucket, 3).await?;
+
+    // List the epochs found in the storage layer
+    let mut epoch_summaries: Vec<EpochSummary> = storage.list_proofs().await?;
+    epoch_summaries.sort_by(|a, b| a.name.epoch.cmp(&b.name.epoch));
+
+    // There should be 3 proofs in the storage layer
+    log::info!(
+        "There are {} epochs in the storage layer",
+        epoch_summaries.len()
+    );
+    assert_eq!(3, epoch_summaries.len());
+
+    // verify all fo the audit proofs
+    for epoch in 0..=2 {
+        let proof_blob = storage.get_proof(epoch).await?;
+        log::info!("Verification epoch {} -> {}", epoch, epoch + 1);
+        crate::auditor::audit_epoch::<Hasher>(proof_blob.clone(), false).await?;
+        crate::auditor::audit_epoch::<Hasher>(proof_blob, true).await?;
```

Comment:
Insanely tiny nitpick: can we assign the bools to variables indicating what they signify or something? It's hard to tell as a reader without seeking out the `audit_epoch` fn separately. Feel free to ignore if you don't find much value in that. Generally, not a major fan of bool arguments (for several reasons) so I may be a bit biased!